server/src/main/java/org/elasticsearch/index/engine/LiveVersionMap.java

```diff
@@ -476,10 +476,18 @@ public long ramBytesUsed() {
     }
 
     /**
-     * Returns how much RAM would be freed up by refreshing. This is the RAM usage of the current version map. It doesn't include tombstones
-     * since they don't get cleared on refresh, nor the old version map that is being reclaimed.
+     * Returns how much RAM is used by refresh. This is the RAM usage of the current and old version maps.
      */
     long ramBytesUsedForRefresh() {
+        return maps.ramBytesUsed();
```

Comment:
I think we should not make this change, unless you have good arguments for it.

The indexing memory controller uses this to figure out whether to flush data and also the order of shards to flush. But the data in `old` is not relevant here, it is memory that is in the process of being released already.

Reply:
@henningandersen this has been answered in the PR. Please take a look.